Foundation.xcodeproj/project.pbxproj

```diff
@@ -2778,7 +2808,7 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/../Frameworks @loader_path/../Frameworks";
 				LIBRARY_SEARCH_PATHS = "$(inherited)";
 				MACH_O_TYPE = mh_execute;
-				OTHER_SWIFT_FLAGS = "-DDEPLOYMENT_ENABLE_LIBDISPATCH -swift-version 3";
+				OTHER_SWIFT_FLAGS = "-DDEPLOYMENT_ENABLE_LIBDISPATCH -swift-version 3 -DTEST_TARGET";
```

Comment:
So that we `import Foundation`/`import SwiftFoundation` here
https://github.com/apple/swift-corelibs-foundation/pull/1048/files#diff-fa28084658370ec9069bda50577d8ab1R14
only when build for **TestFoundation**

`platformConsistentCast` is internal, I cannot expose it to **TestFoundation**, so needed to add this source file to **TestFoundation** target.